src/table_functions.cpp

```diff
@@ -174,23 +209,183 @@ void insert(const CallContext& ctx) {
+    // isn't generated new
+    auto list = ctx.arguments().get(0);
+    auto pos = ctx.arguments().get(1);
+
+    return std::visit(
+        overloaded{
+            [](Table list, Nil /*unused*/) -> Value {
+                auto tmp = list.get(list.border());
+                list.remove(list.border());
+                return tmp;
+            },
+            [&pos](Table list, auto /*pos*/) -> Value {
+                int posi = try_value_is_int(pos, "remove", 2);
+                if (posi > list.border() + 1 || (posi < 1 && list.border() != 0)) {
+                    throw std::runtime_error(
+                        "bad argument #2 to 'remove' (position out of bounds)");
+                }
+                auto tmp = list.get(posi);
+                if (posi == list.border() + 1) {
+                    list.remove(posi);
+                    return tmp;
+                }
+                for (int i = posi + 1; i <= list.border(); i++) {
+                    list.set(posi, list.get(i));
+                    posi++;
+                }
+                list.remove(list.border());
+                return tmp;
+            },
+            [](auto list, auto /*unused*/) -> Value {
+                throw std::runtime_error(
+                    "bad argument #1 to 'remove' (table expected, got " + std::string(list.TYPE) +
+                    ")");
+            }},
+        list.raw(), pos.raw());
+}
+
+void sort(const CallContext& ctx) {
+    auto list = ctx.arguments().get(0);
+    auto comp = ctx.arguments().get(1);
+
+    std::visit(
+        overloaded{
+            [](Table list, Nil /*unused*/) {
+                std::vector<Value> content;
+                for (int i = 1; i <= list.border(); i++) {
+                    content.push_back(list.get(i));
+                }
+                std::sort(content.begin(), content.end(), [](const Value& a, const Value& b) {
+                    return std::get<Bool>(a.less_than(b)).value;
+                });
+                for (int i = 1; i <= list.border(); i++) {
+                    list.set(i, content.at(i - 1));
+                }
+            },
+            [&ctx](Table list, const Function& comp) {
+                std::vector<Value> content;
+                for (int i = 1; i <= list.border(); i++) {
+                    content.push_back(list.get(i));
+                }
+                std::sort(
+                    content.begin(), content.end(),
+                    [&ctx, &comp](const Value& a, const Value& b) -> bool {
+                        auto c = ctx.make_new(Vallist{a, b});
+                        auto erg = comp.call(c).values().get(0);
+
+                        if (erg.type() == "boolean") {
+                            return std::get<Bool>(erg).value;
+                        } else {
+                            throw std::runtime_error("invalid order function for sorting");
+                        }
```

Comment:
Does lua check if there is a bool returned? Usually lua just implicitly converts to a bool when needed, right?

So doing `bool(erg)` might be better. That way all truthy values will become true and all falsy values (only nil and false) will become false.

Reply:
Lua checks if the function returns a boolean. I tested it. 
```lua
function f(a,b)
    return a+b
end

t = {}
for i = 1, 10 do
    t[i] = 100-i
end

table.sort(t,f)
```
with this code the error-message that is listed in the else gets thrown in lua

Reply:
ok then it's fine